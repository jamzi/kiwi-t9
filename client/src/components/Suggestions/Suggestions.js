import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Chip from '@material-ui/core/Chip';

const styles = theme => ({
  paper: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3,
    minHeight: '40px',
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
    justifyContent: 'center'
  }),
  chip: {
    margin: theme.spacing.unit / 2
  }
});

const Suggestions = props => {
  const { classes, words } = props;

  return (
    <Paper className={classes.paper} elevation={4}>
      {words.map((word, index) => {
        return <Chip key={index} label={word} className={classes.chip} />;
      })}
    </Paper>
  );
};

export default withStyles(styles)(Suggestions);
