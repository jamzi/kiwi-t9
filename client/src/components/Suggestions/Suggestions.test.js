import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Suggestions from './Suggestions';

it('renders correctly', () => {
  const mockWordSuggestions = ['tree', 'trie', 'troo'];
  const tree = renderer
    .create(<Suggestions words={mockWordSuggestions} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
