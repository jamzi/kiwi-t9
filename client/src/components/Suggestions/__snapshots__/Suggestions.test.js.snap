// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation4-9 MuiPaper-rounded-4 Suggestions-paper-1"
>
  <div
    className="MuiChip-root-30 Suggestions-chip-2"
    onClick={undefined}
    onKeyDown={[Function]}
    role="button"
    tabIndex={-1}
  >
    <span
      className="MuiChip-label-35"
    >
      tree
    </span>
  </div>
  <div
    className="MuiChip-root-30 Suggestions-chip-2"
    onClick={undefined}
    onKeyDown={[Function]}
    role="button"
    tabIndex={-1}
  >
    <span
      className="MuiChip-label-35"
    >
      trie
    </span>
  </div>
  <div
    className="MuiChip-root-30 Suggestions-chip-2"
    onClick={undefined}
    onKeyDown={[Function]}
    role="button"
    tabIndex={-1}
  >
    <span
      className="MuiChip-label-35"
    >
      troo
    </span>
  </div>
</div>
`;
